docs/client-sdk/javascript/tutorials/use-a-persistent-conversation-cache.md

```diff
@@ -0,0 +1,155 @@
+---
+sidebar_label: Use a persistent conversation cache
+sidebar_position: 7
+---
+
+# Use a persistent conversation cache to improve load time
+
+For a user with a large number of ongoing conversations, loading their conversation list in real-time can be a very expensive operation from the perspective of both bandwidth and CPU usage. This is because the message API client must download all of the [conversation invites](https://xmtp.org/docs/dev-concepts/invitation-and-message-encryption) and decrypt each one using X3DH before being able to display the conversations and their messages.
+
+To improve conversation load time, you can use the XMTP SDK to export the unencrypted conversation list in a JSON serializable format to a persistent conversation cache. For example, you can use the Zustand [`persist` middleware](https://github.com/pmndrs/zustand/blob/main/docs/integrations/persisting-store-data.md) to create a conversation cache that you can persist to the browser’s `LocalStorage`.
+
+:::info
+
+The process described in this tutorial will change soon. You can implement a persistent conversation cache as described here to address performance issues in your app now. However, please be prepared to update your app to use a different approach to be provided in the near future.
+
+:::
+
+When your app creates a new message API client instance, the client can load its internal conversation cache from the browser’s persistent conversation cache.
+
+:::warning
+
+Treat the conversations and data in the cache with the utmost care. Compromise of this data will allow an attacker to impersonate the user on the XMTP network. A compromised script in your app or browser extension could access this data if stored in `LocalStorage`.
+
+:::
+
+## Create a persistent conversation cache
+
+Enable your app to create the persistent conversation cache in its local storage. 
+
+For example, see [`/store/conversationCache.ts`](https://github.com/xmtp-labs/xmtp-inbox-web/blob/main/store/conversationCache.ts) in the example XMTP Inbox web app:
+
+```tsx
+import { create } from "zustand";
+import { persist } from "zustand/middleware";
+import { ConversationExport } from "@xmtp/xmtp-js/dist/types/src/conversations/Conversation";
+import { getEnv } from "../helpers";
+
+export const CONVERSATION_CACHE_VERSION = 1;
```

Comment:
@neekolas removed the separate step to create the constant. Removed the import and added the export const here.